cpu/stm32_common/periph/rtc.c

```diff
@@ -322,4 +326,231 @@ void ISR_NAME(void)
     cortexm_isr_end();
 }
 
+#elif defined(CPU_FAM_STM32F1)
+
+static struct {
+    rtc_alarm_cb_t cb;          /**< callback called from RTC interrupt */
+    void *arg;                  /**< argument passed to the callback */
+} isr_ctx;
+
+static void _rtc_enter_config_mode(void)
+{
+    /* disable backup domain write protection */
+    PWR->CR |= PWR_CR_DBP;
+
+    /* wait until the RTOFF bit is 1 (no RTC register writes ongoing). */
+    while ((RTC->CRL & RTC_CRL_RTOFF) == 0) {}
+
+    /* enter configuration mode. */
+    RTC->CRL |= RTC_CRL_CNF;
+}
+
+static void _rtc_exit_config_mode(void)
+{
+    /* exit configuration mode. */
+    RTC->CRL &= ~RTC_CRL_CNF;
+
+    /* wait until the RTOFF bit is 1 (our RTC register write finished). */
+    while ((RTC->CRL & RTC_CRL_RTOFF) == 0) {}
+
+    /* disable backup domain write protection */
+    PWR->CR &= ~PWR_CR_DBP;
+}
+
+static bool _is_rtc_enable(void)
+{
+    return ((RCC->BDCR & RCC_BDCR_RTCEN) == RCC_BDCR_RTCEN);
+}
+
+static void _rtc_config(void)
+{
+    DEBUG("[RTC] config\n");
+
+    /* enable APB1 clocks */
+    RCC->APB1ENR |= RCC_APB1ENR_PWREN | RCC_APB1ENR_BKPEN;
+
+    /* disable backup domain write protection */
+    PWR->CR |= PWR_CR_DBP;
+
+    /* resets the entire backup domain */
+    RCC->BDCR |= RCC_BDCR_BDRST;
+    /* reset not activated */
+    RCC->BDCR &= ~RCC_BDCR_BDRST;
+
+    /* oscillator clock used as RTC clock */
+    RCC->BDCR |= RCC_BDCR_RTCEN | RCC_BDCR_RTCSEL_LSE;
+
+    /* turn on LSE crystal */
+    RCC->BDCR |= RCC_BDCR_LSEON;
+    while ((RCC->BDCR & RCC_BDCR_LSEON) != RCC_BDCR_LSEON) {}
+
+    /* calibration clock from 0 to 0x7F */
+    BKP->RTCCR |= 0;
+
+    /* second interrupt is disabled. */
+    RTC->CRH &= ~RTC_CRH_SECIE;
+
+    _rtc_enter_config_mode();
+
+    /* if the input clock frequency (fRTCCLK) is 32.768 kHz, write 7FFFh in this register to get a signal period of 1 second. */
+    RTC->PRLH = 0;
+    RTC->PRLL = 0x7FFF;
+
+    _rtc_exit_config_mode();
+
+    /* wait registers synchronize flag */
+    RTC->CRL &= (uint16_t)~RTC_CRL_RSF;
+    while((RTC->CRL & RTC_CRL_RSF) != RTC_CRL_RSF) {}
+
+    /* disable backup domain write protection */
+    PWR->CR &= ~PWR_CR_DBP;
+}
+
+static time_t _rtc_get_time(void)
+{
+    return (RTC->CNTH << 16) | RTC->CNTL;
+}
+
+static void _rtc_set_time(time_t counter_val)
+{
+    _rtc_enter_config_mode();
+    RTC->CNTH = (counter_val & 0xffff0000) >> 16;
+    RTC->CNTL = counter_val & 0x0000ffff;
+    _rtc_exit_config_mode();
+}
+
+void rtc_init(void)
+{
+    /* save current time if RTC already works */
```

Comment:
`rtc_init()` will be called from `periph_init()` implemented in `drivers/periph_common/init.c`. I don't see any reason why it would get called more than once during startup. Therefore, I would suggest to skip the check if the RTC is already running and just configure it here

Reply:
Wakeup from standby would be a reason.

Reply:
Indeed